src/static/index.js

```diff
@@ -123,7 +125,45 @@ firebase.auth().onAuthStateChanged(user => {
     }
 });
 
+let refreshTokenTimer = null;
+
+function clearRefreshToken() {
+    if (refreshTokenTimer !== null) {
+        clearTimeout(refreshTokenTimer);
+        refreshTokenTimer = null;
+    }
+}
+
+function enqueueRefreshToken(user, expirationString) {
+    const expiration = new Date(expirationString).valueOf();
+    const now = new Date().valueOf();
+    const diff = expiration - now;
+
+    clearRefreshToken();
+
+    // Token is already expired, but we just got a token. Something is broken
+    // with Firebase or our clock.  We could refresh the token, but that could
+    // just create an endless loop of errors.  Instead we just bail out.
+    if (diff < 0) {
+        return;
```

Comment:
Her burde det kanskje legges opp til å gi en 401 feil ut til endepunktet som fører til at innlogginsskjermen blir vist på nytt? Hva betyr det f.eks for om du er midt i en oppgave? Burde det være noe retry loops med gradual back off? Dette fungerer sikkert for nå, altså.

Reply:
Tror dette kan gjøres i auth callbacken. Kan se litt på corner caser, men da skal du ha vært offline ganske lenge.

Siden JWT har all data og såvidt jeg vet bruker ikke backend noe ala OCSP/CRL for TLS på disse JWT-tokenene, så er disse gyldige så lenge "exp" ikke har passert. Det er litt av grunnen til å ha en grace periode på 1 minutt.